app/lib/request.rb

```diff
@@ -94,9 +94,16 @@ def http_client
   class Socket < TCPSocket
```

Comment:
@akihikodaki Isn't there different classes for Socket and SSLSocket?

Reply:
`SSLSocket` uses `TCPSocket`, so patching only `TCPSocket` is enough.

Reply:
But this doesn't patch TCPSocket, only our own Socket class, no?

Reply:
`SSLSocket` takes a `TCPSocket` instance as an argument when initializes, and HTTP gem uses our patched `TCPSocket` to create an instance for the argument. That's how it works.

Reply:
Okay, thanks.